### PR TITLE
Feature/improve guilds network management

### LINKED ISSUE #501: Improve network switching in guilds
Currently, guilds network switching has some strange behaviour where the network switching only really works if you connect your wallet to the default network, then switch the connected wallet.

To-do
- Add network to the URL the same as in DXvote
- Fix switching with no connected wallet

Additional related fixes 
- Change default network in guilds to rinkeby temporarily
- FIx ENS PFP loading on networks other than mainnet

### PR DESCRIPTION
**Pull request by mprasanjith**
This introduces a bunch of changes and bugfixes around how network switching works in guilds.

- Handle network switching on the URL
- Fixes to network switching when using injected wallets
- Fix ENS profile picture loading when not on mainnet.
- Add redirect to rinkeby when visiting guilds by default. This is temporary.
- Implement a global error boundary. (We should improve how this looks in the future.)

Fixes #501.

### CODE REVIEW

File: src/components/Guilds/ErrorBoundary/GlobalErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+class GlobalErrorBoundary extends React.Component<{}, { hasError: boolean }> {
```
Comment (hhamud) [inline src/components/Guilds/ErrorBoundary/GlobalErrorBoundary.tsx:3]: Why are we using class style react component here?

File: src/pages/Metadata.tsx [hunk 2]
```diff
@@ -91,7 +91,7 @@ export const CreateMetadataPage = observer(() => {
   } = useContext();
 
   const history = useHistory();
-  const { proposalType } = useParams();
+  const { proposalType } = useParams<{ proposalType?: any }>();
```
Comment (hhamud) [inline src/pages/Metadata.tsx]: Do we not know the ProposalTypes?, best to avoid using any type in general

File: src/hooks/useParams.ts [hunk 3]
```diff
@@ -27,7 +27,7 @@ export const useParams = (searchType: string, defaultSearchtype: string) => {
     history.push({
       location: location.pathname,
       search: `${searchType}=${getParams}`,
-    });
+    } as any);
```
Comment (hhamud) [inline src/hooks/useParams.ts]: I don't understand why this is needed?

File: src/components/Header/index.tsx [hunk 4]
```diff
@@ -49,7 +49,7 @@ const WarningDev = styled.div`
 `;
 
 const Header = observer(() => {
-  const NavItem = withRouter(({ route, history, children }) => {
+  const NavItem = withRouter(({ route, history, children }: any) => {
```
Comment (hhamud) [inline src/components/Header/index.tsx]: is it not possible to avoid using the "any" type?

File: src/components/common/Button.tsx [hunk 5]
```diff
@@ -22,8 +22,8 @@ export const Button = styled.button`
   }
 `;
 
-export const LinkButton = withRouter(
-  ({ route, history, children, ...rest }) => {
+export const LinkButton: any = withRouter(
+  ({ route, history, children, ...rest }: any) => {
```
Comment (hhamud) [inline src/components/common/Button.tsx]: is it not possible to avoid using the "any" type?

**Review by hhamud — CHANGES_REQUESTED** [review]

File: src/components/Guilds/ErrorBoundary/GlobalErrorBoundary.tsx (hunk 1 above)
Comment (mprasanjith) [reply src/components/Guilds/ErrorBoundary/GlobalErrorBoundary.tsx:3]: Weirdly, error boundaries cannot be implemented with functional components as of yet, according to the offical [React docs](https://reactjs.org/docs/error-boundaries.html#introducing-error-boundaries) so had to use a class component here.

**Comment by mprasanjith** [comment]
@hhamud I added the `@types/react-router-dom` library to get types support for React Router and a bunch of typing errors popped up. I didn't wanna add types to the older code so I just used `any`. (Kinda lazy on my part.) I'll add proper types to these now.

File: src/hooks/useParams.ts (hunk 3 above)
Comment (mprasanjith) [reply src/hooks/useParams.ts]: This was throwing a type error and I didn't realize what's wrong here when I typed it as `any`. We have a bug in this code. This should be
```js
history.push({
  pathname: location.pathname,
  search: `${searchType}=${getParams}`,
});
```

It seems we're not using this whole useParams hook though, so I'll remove this whole file.

**Review by rossneilson — DISMISSED** [review]

**Review by hhamud — DISMISSED** [review]

**Review by rossneilson — APPROVED** [review]

**Pull request merged**